02-express-tutorial/app.js

```diff
@@ -1 +1,54 @@
-console.log('Express Tutorial')
+const express = require("express");
+const app = express();
+const { products } = require("./data");
+
+app.listen(3000, () => {
+  console.log("server is listening on port 3000");
+});
+
+app.all("*", (req, res) => {
```

Comment:
Place this code at the end of your file. Otherwise, all requests, including those that should be valid, may return a 'resource not found' error. Express processes routing rules in the order they are defined, and the rule that matches the URL first takes precedence. By placing rules for specific endpoints later in your code, you ensure that the 'resource not found' rule only matches if no other rule matches the requested endpoint.

Reply:
Thank you very much.